src/components/room/widgets/pet-training/PetTrainingPanelWidgetView.tsx

```diff
@@ -0,0 +1,56 @@
+import { DesktopViewEvent, PetTrainingPanelMessageEvent } from '@nitrots/nitro-renderer';
+import { FC } from 'react';
+import { AvatarInfoPet, LocalizeText } from '../../../../api';
+import { Button, Column, Flex, Grid, LayoutPetImageView, NitroCardContentView, NitroCardHeaderView, NitroCardView, Text } from '../../../../common';
+import { useAvatarInfoWidget, useMessageEvent } from '../../../../hooks';
+
+export const PetTrainingPanelWidgetView: FC<{}> = props =>
+{
+    const { avatarInfo = null, petTrainInformation = null, setTrainPetInformation = null } = useAvatarInfoWidget();
+
+    useMessageEvent<DesktopViewEvent>(DesktopViewEvent, event =>
+    {
+        setTrainPetInformation(null);
+    });
+
+    useMessageEvent<PetTrainingPanelMessageEvent>(PetTrainingPanelMessageEvent, event =>
+    {
+        const parser = event.getParser();
+
+        if (!parser) return;
+
+        setTrainPetInformation(parser);
+    });
+
+    const processPetAction = (petId: number, type: string) =>
+    {
+        if (!petId || !type) return;
+
+        // packet for pet actions
```

Comment:
Did you forget to add code here ?

Reply:
I'm looking to find out what it is that does that action.

Reply:
> I'm looking to find out what it is that does that action.

I think it just sends a regular chat message and then the emulator interprets that as a command

Reply:
Yes that's what it does 
![image](https://user-images.githubusercontent.com/25986048/208779784-f50242cb-3a53-4f92-bd46-801173c3cebc.png)

Reply:
This is the code that I have added. It does the speaking correctly when you click on a command, but it doesn't seem to do anything. Perhaps in the emulator it is missing to detect that it is a pet? I do not know, that's the truth. (you can see it in Emulator in `Incoming/Rooms/Users/RoomUserTalkEvent`

![imagen](https://user-images.githubusercontent.com/110488133/208779558-b2004337-3c4e-4635-9a7a-dc1dcb14f648.png)

![imagen](https://user-images.githubusercontent.com/110488133/208779951-c3e869e3-fcc4-44f1-a392-2e6f71ceaf1f.png)

Reply:
> This is the code that I have added. It does the speaking correctly when you click on a command, but it doesn't seem to do anything. Perhaps in the emulator it is missing to detect that it is a pet? I do not know, that's the truth. (you can see it in Emulator in `Incoming/Rooms/Users/RoomUserTalkEvent`
> 
> ![imagen](https://user-images.githubusercontent.com/110488133/208779558-b2004337-3c4e-4635-9a7a-dc1dcb14f648.png)
> 
> ![imagen](https://user-images.githubusercontent.com/110488133/208779951-c3e869e3-fcc4-44f1-a392-2e6f71ceaf1f.png)

I think the emulator isn't detecting it because it only detects english commands. Check your database and see if you have any tables related to pet commands or pet actions

Reply:
ooh!!! It was precisely because of what you just said. I am going to change the client to English so as not to have problems. So everything is ready right?

Reply:
> ooh!!! It was precisely because of what you just said. I am going to change the client to English so as not to have problems. So everything is ready right?

Yeah it will be ready as soon as you commit this last change 😆

Reply:
Perfect !!! Right now I do.

Thank you very much for the information and for the work we have done as a team !!! 👏🏼 